projects/netgrif-components-core/src/lib/panel/task-panel-single/abstract-single-task.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, EventEmitter, Inject, Input, OnDestroy, Optional, Output, TemplateRef } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { TaskPanelData } from '../task-panel-list/task-panel-data/task-panel-data';
+import { MatExpansionPanel } from '@angular/material/expansion';
+import { HeaderColumn } from '../../header/models/header-column';
+import { TaskEventNotification } from '../../task-content/model/task-event-notification';
+import { LoggerService } from '../../logger/services/logger.service';
+import { NAE_TAB_DATA } from '../../tabs/tab-data-injection-token/tab-data-injection-token';
+import { InjectedTabData } from '../../tabs/interfaces';
+import { ActivatedRoute } from '@angular/router';
+
+
+@Component({
+    selector: 'ncc-abstract-single-task',
+    template: ''
+})
+export abstract class AbstractSingleTaskComponent implements OnDestroy {
+
+    protected _task$: Observable<TaskPanelData>;
+    protected _unsubscribe$: Subject<void>;
+    protected _taskPanelRef: MatExpansionPanel;
+    @Input() loading$: Observable<boolean>;
+    @Input() selectedHeaders$: Observable<Array<HeaderColumn>>;
+    @Input() responsiveBody = true;
```

Comment:
Looks like too many inputs, maybe create a configuration class?

Reply:
corrected

Reply:
After creating configuration class it resulted in too long checks for existence of that class and its fields in HTML, consider keep it as is?